### DISCUSSION TITLE
CONSTRUCT query response with prefixes

### DISCUSSION THREAD

**Discussion by mathib** [comunica/comunica#894] [Q&A] [answered]
When doing a CONSTRUCT query with content type set to `text/turtle`, there appears to be <some> function to provide the prefixes that should be used in the CONSTRUCT query response, but I can't find how. Adding prefixes inside the CONSTRUCT query does not change the output.

The Comunica widget has support for prefixes in the RDF output as I noticed [here](http://query.linkeddatafragments.org/#datasources=%2F%2Ffragments.dbpedia.org%2F2016-04%2Fen&query=CONSTRUCT%20%7B%20%20%20%3Fmovie%20dbpedia-owl%3Astarring%20%5B%20rdfs%3Alabel%20%22Brad%20Pitt%22%40en%20%5D%3B%0A%20%20%20%20%20%20%20%20%20rdfs%3Alabel%20%3Ftitle%3B%0A%20%20%20%20%20%20dbpedia-owl%3Adirector%20%5B%20rdfs%3Alabel%20%3Fname%20%5D.%20%7D%0AWHERE%20%7B%0A%20%20%3Fmovie%20dbpedia-owl%3Astarring%20%5B%20rdfs%3Alabel%20%22Brad%20Pitt%22%40en%20%5D%3B%0A%20%20%20%20%20%20%20%20%20rdfs%3Alabel%20%3Ftitle%3B%0A%20%20%20%20%20%20%20%20%20dbpedia-owl%3Adirector%20%5B%20rdfs%3Alabel%20%3Fname%20%5D.%0A%20%20FILTER%20LANGMATCHES(LANG(%3Ftitle)%2C%20%22EN%22)%0A%20%20FILTER%20LANGMATCHES(LANG(%3Fname)%2C%20%20%22EN%22)%0A%7D).

**Comment by rubensworks** [comment, marked as answer]
Currently, this is [a config-level setting](https://github.com/comunica/comunica/blob/master/packages/actor-sparql-parse-algebra/components/Actor/SparqlParse/Algebra.jsonld#L31-L48) in Comunica. There currently isn't a way to change this programmatically, other than configuring a custom Comunica instance.

**Comment by danielbakas** [comment]
Is there a way to pass a context to a query (much like the `@context` for JSON-LD) to construct the prefixes in a `text/turtle` query, and therefore make it more compact/readable?

**Reply by rubensworks** [reply]
No, it's not possible to use `@context` with prefixes for SPARQL queries.

**Reply by danielbakas** [reply]
Yet... 😏😄
